### PR TITLE
game-porting-toolkit: New port

### PR DESCRIPTION
**Pull request by Gcenx**
#### Description

From Apples Developers site:

> Use the game porting toolkit to eliminate months of up-front work and evaluate how well your game could run on Mac before writing any code.

<br>

A macports equivalent to https://github.com/apple/homebrew-apple/blob/main/Formula/game-porting-toolkit.rb with some modifications to avoid as many manual steps as possible.

This port is the `wine` portion the other part is the new `d3dmetal` port that's responsible for installing Apples Direct3D Metal, that requires a manual download.

<br>

_Please note this is only intended for use on Apple Silicon systems, normal `wine` ports will follow_

###### Type(s)

- [x] New submission

###### Tested on

macOS 14.5 23F79 x86_64
Xcode 15.4 15F31d

###### Verification 

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: game-porting-toolkit/Portfile [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+post-destroot {
+    ln -sf ${prefix}/share/wine/gecko ${destroot}${prefix}/libexec/${name}/share/wine/gecko
+    ln -sf ${prefix}/share/wine/mono ${destroot}${prefix}/libexec/${name}/share/wine/mono
+
+    file copy ${destroot}${prefix}/libexec/${name}/bin/wine64           ${destroot}${prefix}/libexec/${name}/bin/wine
+    file copy ${destroot}${prefix}/libexec/${name}/bin/wine64-preloader ${destroot}${prefix}/libexec/${name}/bin/wine-preloader
+
+    file copy ${filespath}/gameportingtoolkit           ${destroot}${prefix}/bin/gameportingtoolkit
+    file copy ${filespath}/gameportingtoolkit-no-esync  ${destroot}${prefix}/bin/gameportingtoolkit-no-esync
+    file copy ${filespath}/gameportingtoolkit-no-hud    ${destroot}${prefix}/bin/gameportingtoolkit-no-hud
```
Comment (ryandesign) [inline game-porting-toolkit/Portfile:231]: Can be simplified to:
```suggestion
    xinstall -W ${filespath} \
        gameportingtoolkit \
        gameportingtoolkit-no-esync \
        gameportingtoolkit-no-hud \
        ${destroot}${prefix}/bin/
```

File: game-porting-toolkit/Portfile [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+
+    reinplace -q "s|@@PREFIX@@|${prefix}|g"             ${destroot}${prefix}/bin/gameportingtoolkit \
+                                                        ${destroot}${prefix}/bin/gameportingtoolkit-no-esync \
+                                                        ${destroot}${prefix}/bin/gameportingtoolkit-no-hud
```
Comment (ryandesign) [inline game-porting-toolkit/Portfile:235]: The `-q` flag should only be used when you want MacPorts not to notify you if a no replacement took place. The only reason you might want to do this is if you are batch-processing all files of a particular type and not all of them need the replacement; that's not the case here as you're listing three specific files.

Can be simplified to:

```suggestion
    reinplace -W ${destroot}${prefix}/bin "s|@@PREFIX@@|${prefix}|g" \
        gameportingtoolkit \
        gameportingtoolkit-no-esync \
        gameportingtoolkit-no-hud
```

File: game-porting-toolkit/files/gameportingtoolkit [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+if [ -z "$1" ];  then
+	echo "Usage: $0 <wine-prefix-path> <executable>"
+fi
+
+exe_path="cmd.exe"
+if [ ! -z "$2" ]; then
+	exe_path="$2"
+fi
+
+WINEDLLPATH_PREPEND="@@PREFIX@@/libexec/d3dmetal/wine" MTL_HUD_ENABLED=1 WINEESYNC=1 WINEPREFIX="$1" @@PREFIX@@/libexec/game-porting-toolkit/bin/wine64 "$exe_path" 2>&1 | grep "D3DM"
```
Comment (ryandesign) [inline game-porting-toolkit/files/gameportingtoolkit:12]: Why is the stderr of `wine64` being redirected to stdout and why is that stdout being piped through `grep "D3DM"`? Why do you want to throw away all of wine's other output?

File: game-porting-toolkit/files/gameportingtoolkit [hunk 4]
```diff
@@ -0,0 +1,12 @@
+#!/bin/zsh
+
+if [ -z "$1" ];  then
+	echo "Usage: $0 <wine-prefix-path> <executable>"
+fi
```
Comment (ryandesign) [inline game-porting-toolkit/files/gameportingtoolkit:5]: Did you mean to `exit` after printing this usage message?

**Comment by pmetzger** [comment]
Is this now good to go?

**Comment by pmetzger** [comment]
@Gcenx If @ryandesign doesn't get back to you in a few days, ping me to merge this.

**Comment by ryandesign** [comment]
Nobody checked the CI logs before merging? This was added in the wrong directory so CI never saw it as a port that needed to be built.

**Comment by pmetzger** [comment]
@ryandesign The CI passed. This means we have a bug in the CI scripts. We depend on the green checkmark to tell us that things worked; we don't read the logs if it passes.

**Comment by ryandesign** [comment]
CI also passes if the port does not support the architecture being tested.

You must check the logs. A green check mark is not sufficient. At least check that the CI phase that is supposed to build the port took more than zero seconds.

**Comment by pmetzger** [comment]
These builds took more than zero seconds. It's hard to be an expert on every port. One relies to a considerable extent on the port maintainer.

If it is truly the case that one needs to hand-check all the CI logs, then I don't think I would be able to merge pull requests any longer. I cannot find time for that on top of trying to understand the changes being made themselves. I merge large numbers of pull requests, partially because it seems like few other people bother to do it, but I don't even really have time for that in my day and other things I do are suffering for it.

If passing the CI does not actually mean that the CI passes and I have to read the logs for every CI run too, then the CI doesn't do me any good, and I really don't have time to do more.

**Pull request merged**